modules/calc.py

```diff
@@ -20,30 +21,55 @@
     (r'\/', '/'),
 ]
 
+r_google_calc = re.compile(r'calculator-40.gif.*? = (.*?)<')
+r_google_calc_exp = re.compile(r'calculator-40.gif.*? = (.*?)<sup>(.*?)<')
 
 def c(phenny, input):
-    """DuckDuckGo calculator."""
+    """Google calculator."""
     if not input.group(2):
         return phenny.reply("Nothing to calculate.")
     q = input.group(2)
-
-    try:
-        r = web.get(
-            'https://api.duckduckgo.com/?q={}&format=json&no_html=1'
-            '&t=mutantmonkey/phenny'.format(web.quote(q)))
-    except web.HTTPError:
-        raise GrumbleError("Couldn't parse the result from DuckDuckGo.")
-
-    data = web.json(r)
-    if data['AnswerType'] == 'calc':
-        answer = data['Answer'].split('=')[-1].strip()
+    bytes = generic_google(q)
+    m = r_google_calc_exp.search(bytes)
+    if not m:
+        m = r_google_calc.search(bytes)
+   
+    if not m:
+        num = None
+    elif m.lastindex == 1:
+        num = web.decode(m.group(1))
     else:
-        answer = None
+        num = "^".join((web.decode(m.group(1)), web.decode(m.group(2))))
 
-    if answer:
-        phenny.say(answer)
+    if num:
+        phenny.say(num)
     else:
-        phenny.reply('Sorry, no result.')
+        phenny.reply("Sorry, no result.")
+
+
+# def c(phenny, input):
```

Comment:
Please remove this old code. It's no problem to pull stuff from the Git history if it turns out that we need it later.